# Lambda School Data Science - Unit 1 Sprint 2 Module 2

---

## Assignment: Hypothesis Testing (chi-square tests)


# Objectives

* Objective 01 - explain the purpose of a chi-square test and identify applications
* Objective 02 - set up a chi-square test for independence on two categorical variables
* Objective 03 - use a chi-square test p-value to draw the correct conclusion about the null and alternative hypothesis
* Objective 04 - explain how to compare the null hypothesis and sample and how randomness in the sampling process affect the result





### Dataset description:

Slasher movies thrill audiences by portraying a lone antagonist (typically male) who attacks innocent victims with extreme violence and without apparent motive. However, this exciting (if gory) subgenre of horror film is criticized by those who view the violence as being used to “punish” female characters who engage in sexual activity during the film. 

To test this claim, study authors randomly sampled 50 North American slasher films released from 1960 to 2009 and coded the 485 characters appearing in them as being male or female, involved in sexual activity or not, and if they survived the film or not. 

The data appears in the Slasher.csv dataset. In this project, we are going to answer a slightly simpler question: Are female characters in slasher films more likely to be "murdered" than male characters?

Source: A. Welsh (2010). "On the Perils of Living Dangerously in the Slasher Horror Film: Gender Differences in the Association Between Sexual Activity and Survival," Sex Roles, Vol. 62 pp. 762-773

You can find the Slasher.csv file here:

<https://raw.githubusercontent.com/Chelsea-Myers/Lambda-Intro/master/Slasher.csv>



### 1) Load the Slasher.csv dataset in a new dataframe. 

* Call your DataFrame `slasher_df`
* set `skipinitialspace=True`
* set `header=0`


In [1]:
# Imports (we won't provide these for every assignment!)
import pandas as pd
import numpy as np

# Load the dataset
slasher_df = pd.read_csv('https://raw.githubusercontent.com/Chelsea-Myers/Lambda-Intro/master/Slasher.csv', skipinitialspace=True, header=0)
### your code here ###


### Answer checks

We're going to continue using the assert statements we used in the Module 1 Project (assignment) to check our work. The cells with the `assert` statements can be skipped or deleted but you should try to leave them in. Remember - it's a way to check your work as you go through this Module Project and also get some feedback if you have an error.

###2) Display the top 5 rows of the dataframe to take a look at the variable names and values.

In [15]:
### your code here ###
slasher_df.head(20)

ID  Gender  Activity  Survival
0    1       1         1         1
1    2       1         1         1
2    3       1         1         1
3    4       1         1         1
4    5       1         1         1
5    6       1         1         1
6    7       1         1         1
7    8       1         1         1
8    9       1         1         1
9   10       1         1         1
10  11       1         1         1
11  12       1         1         0
12  13       1         1         0
13  14       1         1         0
14  15       1         1         0
15  16       1         1         0
16  17       1         1         0
17  18       1         1         0
18  19       1         1         0
19  20       1         1         0

In [3]:
# This is an ANSWER CHECK cell.
# Don't alter this cell if you want accurate feedback
#------------------------------------------------------------------------------#

# check if the DataFrame was created
assert not slasher_df.empty, 'Make sure the df name is accurate and you loaded the correct URL.'
# check the shape of the DataFrame
assert slasher_df.shape == (485, 4), 'Is your data loaded with the correct argument?'
print('Correct! Continue to the next question')

Correct! Continue to the next question


In this dataset, the variable codes are:

* ID = Character ID
* Gender = Character gender.  0 = male, 1 = female
* Activity = Character was portrayed as engaged in sexual activity.  0 = no, 1 = yes
* Survival = Character survival in the film. 0 = did not survive, 1 = survived

###3) Calculate the frequency and relative frequency of male and female characters in this set of slasher films.  Describe your results in a sentence or two and use your results to answer the question below.

In [4]:
### your code here ###
slasher_df['Gender'].value_counts()


0    263
1    222
Name: Gender, dtype: int64

In [8]:
female_ratio = round(222/ 485 * 100)
female_ratio

46

Males: 263
Females: 222
Total = 485

QUIZ: What percentage of the characters are **female** (gender = 1)? Run the cell below, and enter your percentage (*rounded to the nearest whole number*). Then run the QUIZ CHECK cell.

---



In [7]:
# This is an QUIZ cell. Run the cell, enter your answer.
# Don't alter this cell if you want accurate feedback
#------------------------------------------------------------------------------#
female_percent = input()

46


In [9]:
# QUIZ CHECK cell
#------------------------------------------------------------------------------#
assert female_percent == str(46), 'Double check your percentages.'
print('Correct! Continue to the next question.')

Correct! Continue to the next question.


SHORT ANSWER TEXT HERE

###4) Calculate the frequency and relative frequency of character survival in this set of slasher films.  Describe your results in a sentence or two.

In [10]:
### your code here ###
slasher_df['Survival'].value_counts()

0    400
1     85
Name: Survival, dtype: int64

SHORT ANSWER TEXT HERE

Did not Survive: 400
Survived: 85
Total = 485


In [11]:
died = 400 / 485 * 100
died

survived = round(85/ 485 * 100)
survived

18

QUIZ: What percentage of characters *survived*? Run the cell below, enter your percentage (*rounded to the nearest whole number*).

In [12]:
# This is an QUIZ cell. Run the cell, enter your answer.
# Don't alter this cell if you want accurate feedback
#------------------------------------------------------------------------------#
percent_survival = input()

18


In [13]:
# QUIZ CHECK cell
#------------------------------------------------------------------------------#
assert percent_survival == str(18), 'Double check your percentages.'
print('Correct! Continue to the next question.')

Correct! Continue to the next question.


### 5) Calculate the joint distribution of character sex and movie survival. 

Hint: If you are working from the in-class examples, it will be helpful to include character sex as the index variable. 

Describe your results in a sentence or two.

In [29]:
# Calculate the joint distribution

### your code here ###

x =pd.crosstab(slasher_df['Gender'], slasher_df['Survival'])

### 5b) Do the same as 5a) but **add the table margins**.  Assign this to a variable called `joint_dist_margins`.

In [32]:
# Add the table margins to the joint distribution

### your code here ###
joint_dist_margins = pd.crosstab(slasher_df['Gender'], slasher_df['Survival'], margins=True)
joint_dist_margins

Survival    0   1  All
Gender                
0         228  35  263
1         172  50  222
All       400  85  485

In [33]:
from scipy import stats

In [34]:
table = np.array([[228,35], [172, 50]])
stats.chi2_contingency(table, correction=False)

(7.07161567740857,
 0.007831460362307106,
 1,
 array([[216.90721649,  46.09278351],
        [183.09278351,  38.90721649]]))

In [22]:
# This is an ANSWER CHECK cell.
# Don't alter this cell if you want accurate feedback
#------------------------------------------------------------------------------#

# check if the DataFrame was created
assert not joint_dist_margins.empty, 'Did you create your DataFrame with the joint distribution?'
# check the shape of the DataFrame
assert joint_dist_margins.shape == (3, 3), 'Did you include the margins?'
print('Correct! Continue to the next question.')

Correct! Continue to the next question.


SHORT ANSWER TEXT HERE

###6) Calculate the distribution of survival conditional on character sex.
* assign the crosstab DataFrame to the variable `slasher_survival`
* make sure to normalize on the "index"
* multiply the resulting DataFrame by 100 to return answers as percents

Hint: If you are working from the in-class examples, it will be helpful to include character sex as the index variable. 

Describe your results in a sentence or two in the cell below.

In [35]:
# Create your slasher_survival DataFrame here

### your code here ###
slasher_survival = x
slasher_survival

Survival    0   1
Gender           
0         228  35
1         172  50

In [74]:
rows = slasher_survival.iloc[0:2].values
rows_sum = sum(rows)
# rows_sum

# cols = slasher_survival


In [85]:
import math

In [95]:
expected = []
for i in range(len(rows)):
  er = []
  for col in rows[i]:
    val = round(col / sum(rows[i])* 100,)
    er.append(val)
  expected.append(er)
expected = np.array(expected)
expected

array([[87, 13],
       [77, 23]])

In [96]:
slasher_survival = pd.DataFrame(expected)

In [97]:
# This is an ANSWER CHECK cell.
# Don't alter this cell if you want accurate feedback
#------------------------------------------------------------------------------#

# check if the DataFrame was created
assert not slasher_survival.empty, 'Did you calculate the distribution?'
# check the shape of the DataFrame
assert round(slasher_survival[0][0]) == 87.0, 'Is your DataFrame normalized and converted to a percent?'
print('Correct! Continue to the next question.')

Correct! Continue to the next question.


SHORT ANSWER TEXT HERE

###7) Identify the appropriate null and alternative hypotheses to test for a statistically significant relationship between character sex and survival. 

**Ho:** There is no relationship between character sex and movie survival.

**Ha:**. There is a relationship between character sex and movie survival.

### 8) Perform the chi-square test above using `stats.chi2_contingency()`. The docs for this function are found [here](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.chi2_contingency.html).

* assign the p-value to the variable `p_val`
* assign the other variables to whatever names you prefer


In [100]:
# Perform the chi-square test

### your code here ###

chi2_stat , p_val, dof, expected = stats.chi2_contingency(table)
p_val

0.011104692591139975

In [101]:
# This is an ANSWER CHECK cell.
# Don't alter this cell if you want accurate feedback
#------------------------------------------------------------------------------#

# check the p_val
assert round(p_val, 2) == 0.01, 'Double-check the variables you returned; p_val should be the second in the list'
print('Correct! Continue to the next question.')

Correct! Continue to the next question.


### 9) Interpret your p-value in 8) using the alpha = 0.05 significance level, and summarize your results in a sentence or two.

SHORT ANSWER TEXT HERE

### 10) Create a side-by-side bar plot to illustrate the relationship between character sex and survival in slasher movies.  Sum up your overall results in a sentence or two.

In [ ]:
# Plot time!
import matplotlib.pyplot as plt

## your code here ##

Female characters are more likely to survive a slasher film than male characters, but both (unsuprisingly) don't have a good survival rate.